airbyte-integrations/connectors/source-mongodb-v2/src/main/java/io/airbyte/integrations/source/mongodb/state/MongoDbStateManager.java

```diff
@@ -206,4 +226,97 @@ private boolean isValidStreamDescriptor(final StreamDescriptor streamDescriptor)
     }
   }
 
+  // Required when using stateIterator related functions below.
+  public void withIteratorFields(
```

Comment:
Can we just add this to the createStateManager method out of curiosity?

https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connectors/source-mongodb-v2/src/main/java/io/airbyte/integrations/source/mongodb/state/MongoDbStateManager.java#L48